connector/src/it/scala/com/datastax/spark/connector/datasource/CassandraCatalogNamespaceSpec.scala

```diff
@@ -0,0 +1,113 @@
+
+  }
+
+  it should "list a keyspaces in a keyspace"  in {
+    spark.sql(s"CREATE DATABASE IF NOT EXISTS $defaultKs WITH DBPROPERTIES (class='SimpleStrategy',replication_factor='5')")
+    waitForKeyspaceToExist(defaultKs, true)
+    //Using Catalog and DefaultKS triggers listNamespaces(namespace) pathway
+    val result = spark.sql(s"SHOW NAMESPACES FROM $catalogName.$defaultKs").collect
+    result shouldBe empty
+  }
+
+  it should "give no keyspace found when looking up a keyspace which doesn't exist" in {
+    intercept[NoSuchNamespaceException]{
+      spark.sql(s"SHOW NAMESPACES FROM $catalogName.fakenotreal").collect
+    }
+  }
+
+  it should "be able to create a new keyspace" in {
+    dropKeyspace(defaultKs)
+    waitForKeyspaceToExist(defaultKs, false)
+    spark.sql(s"CREATE DATABASE $defaultKs WITH DBPROPERTIES (class='SimpleStrategy',replication_factor='5')")
+    waitForKeyspaceToExist(defaultKs, true)
+    spark.sql(s"DESCRIBE DATABASE EXTENDED $defaultKs").show
+  }
+
+  it should "be able to create a new keyspace with NTS" in {
+    dropKeyspace(defaultKs)
+    waitForKeyspaceToExist(defaultKs, false)
+    val datacenterName = getMetadata().getNodes.values().iterator().next().getDatacenter
+    spark.sql(s"CREATE DATABASE IF NOT EXISTS $defaultKs WITH DBPROPERTIES (class='NetworkTopologyStrategy', $datacenterName='1')")
+    waitForKeyspaceToExist(defaultKs, true)
+    spark.sql(s"DESCRIBE DATABASE EXTENDED $defaultKs").show
+  }
+
+  it should "give a sensible error when trying to create a keyspace without a replication class" in {
+    dropKeyspace(defaultKs)
+    waitForKeyspaceToExist(defaultKs, false)
+    val exception = intercept[CassandraCatalogException]{
+      spark.sql(s"CREATE DATABASE $defaultKs WITH DBPROPERTIES (replication_factor='5')")
+    }
+    exception.getMessage should include("requires a class")
+  }
+
+  it should "give a sensible error when trying to create a keyspace with an invalid replication strategy" in {
+    dropKeyspace(defaultKs)
+    waitForKeyspaceToExist(defaultKs, false)
+    val exception = intercept[CassandraCatalogException]{
+      spark.sql(s"CREATE DATABASE $defaultKs WITH DBPROPERTIES (class='foobar',replication_factor='5')")
+    }
+    exception.getMessage should include("Unknown keyspace replication strategy")
+  }
+
+  it should "throw a keyspace exists exception when creating a keyspace that already exists" in {
+    spark.sql(s"CREATE DATABASE IF NOT EXISTS $defaultKs WITH DBPROPERTIES (class='SimpleStrategy',replication_factor='5')")
+    intercept[NamespaceAlreadyExistsException]{
+      spark.sql(s"CREATE DATABASE $defaultKs WITH DBPROPERTIES (class='SimpleStrategy',replication_factor='5')")
+    }
+  }
+
+  it should "explain Cassandra specific properties in the describe statement" in {
+    spark.sql(s"CREATE DATABASE IF NOT EXISTS $defaultKs WITH DBPROPERTIES (class='SimpleStrategy',replication_factor='5')")
```

Comment:
Nit: don't we want the dropKeyspace() + waitForKeyspaceToExist(false) sequence here as well?

Reply:
sure, this would break if another keyspace with different properties was made, I wrote this before I had the NTS test in there too so all the keyspaces were the same anyway